adapter/internal/adapter/adapter.go

```diff
@@ -308,7 +308,7 @@ func fetchAPIsOnStartUp(conf *config.Config) {
 				logger.LoggerMgw.Debugf("Time Duration for retrying: %v", conf.ControlPlane.RetryInterval*time.Second)
 				time.Sleep(conf.ControlPlane.RetryInterval * time.Second)
 				logger.LoggerMgw.Infof("Retrying to fetch API data from control plane.")
-				synchronizer.FetchAPIs(&d.APIUUID, d.GatewayLabels, c)
+				go synchronizer.FetchAPIs(&d.APIUUID, d.GatewayLabels, c)
```

Comment:
Ready state is set from this function, please check the readiness probes are working in k8s without any issue

Reply:
Could you please explain why we need a go-routine here?

Reply:
During the startup if this happens in the main thread, it can cause adapter to crash, isn't it. So we thought of taking this out to separate routine so the adapter will not crash if error during pulling the APIs. Only the ready state will not be set.

Reply:
Here too, this goroutine is already under an existing goroutine. So, it won't end up crashing. The root cause for the crash that we observed was the `Fatal` log which essentially does an os.Exit(1). I think we can remove this too.

Reply:
If at the root level we use  routine then here we don't need I think

Reply:
Will remove this goroutine in that case